fedimint-client/src/module/init/recovery.rs

```diff
@@ -247,24 +247,32 @@ where
             api: DynGlobalApi,
             decoders: ModuleDecoderRegistry,
             epoch_range: ops::Range<u64>,
-        ) -> impl futures::Stream<Item = (u64, SessionOutcome)> + 'a {
+        ) -> impl futures::Stream<Item = (u64, Vec<AcceptedItem>)> + 'a {
             // How many request for blocks to run in parallel (streaming).
             const PARALLISM_LEVEL: usize = 8;
 
-            futures::stream::iter(epoch_range)
-                .map(move |block_idx| {
+            futures::stream::iter(epoch_range.clone())
+                .map(move |session_idx| {
                     let api = api.clone();
                     let decoders = decoders.clone();
                     Box::pin(async move {
-                        info!(block_idx, "Fetching epoch");
+                        info!(session_idx, "Fetching epoch");
 
                         let mut retry_sleep = Duration::from_millis(10);
                         let block = loop {
-                            info!(target: LOG_CLIENT_RECOVERY, block_idx, "Awaiting signed block");
-                            match api.await_block(block_idx, &decoders).await {
```

Comment:
We cannot switch away from that API *in general*. I'd propose using `await_block` for API `0.0` (causing recovery to take longer) while using the new API for `0.1` so we can deprecate the `await_block` endpoint longer-term (should add a FIXME annotation).

For API version determination we can likely add a cached API call to `DynGlobalAPI`.

Great to see the compatibility tests catching this @bradleystachurski!

Reply:
Oh, right. Forgot about it. Again.